devenv/sync.py

```diff
@@ -30,6 +30,8 @@ def main(context: Context, argv: Sequence[str] | None = None) -> int:
         "repo": repo.name,
         "coderoot": context.get("code_root"),
     }
+
+    os.chdir(repo.path)
     return module.main(context_compat)  # type: ignore
```

Comment:
I would:

```python
with contextlib.chdir(repo.path):
    return module.main(...)
```

that way it is properly undone if a test calls this (should we add a test to demo this behaviour?)

Reply:
oooh that's good